Core/include/Acts/Navigation/NavigationDelegate.hpp

```diff
@@ -0,0 +1,36 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Surfaces/BoundaryTolerance.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+
+namespace Acts {
+
+class NavigationStream;
+class Logger;
+
+/// Struct that serves as the argument to the navigation delegate.
+/// It is not supposed to be used as an lvalue.
+struct NavigationArguments {
+  NavigationStream& main;
```

Comment:
this object will not be copy or movable, but I guess we don't need that?

Reply:
I'm thinking of this like a parameter pack, so that we don't have to put everything into the signature of the navigation call itself.